prime-dotnet-webapi/Services/CommunitySiteService.cs

```diff
@@ -194,42 +194,45 @@ private void UpdateBusinessHours(Site current, CommunitySiteUpdateModel updated)
             }
         }
 
-        private void UpdateRemoteUsers(Site current, CommunitySiteUpdateModel updated)
+        private void UpdateRemoteUsers(Site current, IEnumerable<RemoteUser> updateRemoteUsers)
         {
-            // Wholesale replace the remote users
-            foreach (var remoteUser in current.RemoteUsers)
+            if (updateRemoteUsers == null)
             {
-                foreach (var certification in remoteUser.RemoteUserCertifications)
-                {
-                    _context.Remove(certification);
-                }
-                _context.RemoteUsers.Remove(remoteUser);
+                return;
             }
 
-            if (updated?.RemoteUsers != null && updated?.RemoteUsers.Count() != 0)
+            foreach (var updateRemoteUser in updateRemoteUsers)
             {
-                foreach (var remoteUser in updated.RemoteUsers)
-                {
-                    remoteUser.SiteId = current.Id;
-                    var remoteUserCertifications = new List<RemoteUserCertification>();
+                var existingRemoteUser = _context.RemoteUsers
+                    .SingleOrDefault(u => u.Id == updateRemoteUser.Id);
 
-                    foreach (var certification in remoteUser.RemoteUserCertifications)
+                if (existingRemoteUser == null)
+                {
+                    updateRemoteUser.Id = 0;
+                    current.RemoteUsers.Add(updateRemoteUser);
+                }
+                else
+                {
+                    updateRemoteUser.SiteId = current.Id;
+                    _context.Entry(existingRemoteUser).CurrentValues.SetValues(updateRemoteUser);
+                    foreach (var certification in existingRemoteUser.RemoteUserCertifications)
                     {
-                        var newCertification = new RemoteUserCertification
+                        var existingCertification = existingRemoteUser.RemoteUserCertifications
+                            .FirstOrDefault(c => c.Id == certification.Id);
```

Comment:
This line does nothing, it takes the certification and then finds it again?

Reply:
Good catch. Should be ` var existingCertification = _context.RemoteUserCertifications...`

Reply:
We don't have to update the `RemoteUserCertification`s, we can just drop and add them

Reply:
> Code is not working as intended
> 
> ![image](https://user-images.githubusercontent.com/39168456/140827315-458b96b4-890e-43b3-9600-4de4b5f2297a.png) ![image](https://user-images.githubusercontent.com/39168456/140827240-47b7092b-ebf1-476c-9c83-1d87b626f072.png)
> 
> ![image](https://user-images.githubusercontent.com/39168456/140827489-ce70765a-518a-4f6f-a468-dab86deb955d.png) ![image](https://user-images.githubusercontent.com/39168456/140827553-74d935b9-aac8-4f52-8328-ec41dd7fc9c6.png)

Saw the unit test. Updated code.